velox/common/base/PeriodicStatsReportDaemon.cpp

```diff
@@ -0,0 +1,348 @@
+        memoryCacheStats.ssdStats->growFileErrors);
+    RECORD_METRIC_VALUE(
+        kMetricSsdCacheCumulativeWriteSsdErrors,
+        memoryCacheStats.ssdStats->writeSsdErrors);
+    RECORD_METRIC_VALUE(
+        kMetricSsdCacheCumulativeWriteCheckpointErrors,
+        memoryCacheStats.ssdStats->writeCheckpointErrors);
+    RECORD_METRIC_VALUE(
+        kMetricSsdCacheCumulativeReadSsdErrors,
+        memoryCacheStats.ssdStats->readSsdErrors);
+    RECORD_METRIC_VALUE(
+        kMetricSsdCacheCumulativeReadCheckpointErrors,
+        memoryCacheStats.ssdStats->readCheckpointErrors);
+    RECORD_METRIC_VALUE(
+        kMetricSsdCacheCachedEntries, memoryCacheStats.ssdStats->entriesCached);
+    RECORD_METRIC_VALUE(
+        kMetricSsdCacheCachedRegions, memoryCacheStats.ssdStats->regionsCached);
+    RECORD_METRIC_VALUE(
+        kMetricSsdCacheCachedBytes, memoryCacheStats.ssdStats->bytesCached);
+    REPORT_IF_NOT_ZERO(
+        kMetricSsdCacheCheckpointsRead,
+        memoryCacheStats.ssdStats->checkpointsWritten -
+            prevSimpleStatsHolder_[kLastSsdCacheCheckpointsWritten].int64Value);
+    REPORT_IF_NOT_ZERO(
+        kMetricSsdCacheCheckpointsWritten,
+        memoryCacheStats.ssdStats->checkpointsRead -
+            prevSimpleStatsHolder_[kLastSsdCacheCheckpointsRead].int64Value);
+
+    prevSimpleStatsHolder_[kLastSsdCacheCheckpointsWritten].int64Value =
+        memoryCacheStats.ssdStats->checkpointsWritten;
+    prevSimpleStatsHolder_[kLastSsdCacheCheckpointsRead].int64Value =
+        memoryCacheStats.ssdStats->checkpointsRead;
+  }
+
+  if (auto* cacheTTLController =
+          velox::cache::CacheTTLController::getInstance()) {
+    RECORD_METRIC_VALUE(
+        kMetricCacheMaxAgeSecs,
+        cacheTTLController->getCacheAgeStats().maxAgeSecs);
+
+    RECORD_METRIC_VALUE(
+        kMetricMemoryCacheNumAgedOutEntries,
+        memoryCacheStats.numAgedOut -
+            prevSimpleStatsHolder_[kLastMemoryCacheAgedOuts].int64Value);
+    prevSimpleStatsHolder_[kLastMemoryCacheAgedOuts].int64Value =
+        memoryCacheStats.numAgedOut;
+    RECORD_METRIC_VALUE(
+        kMetricMemoryCacheNumCumulativeAgedOutEntries,
+        memoryCacheStats.numAgedOut);
+
+    if (memoryCacheStats.ssdStats != nullptr) {
+      RECORD_METRIC_VALUE(
+          kMetricSsdCacheCumulativeAgedOutEntries,
+          memoryCacheStats.ssdStats->entriesAgedOut)
+      RECORD_METRIC_VALUE(
+          kMetricSsdCacheCumulativeAgedOutRegions,
+          memoryCacheStats.ssdStats->regionsAgedOut);
+    }
+  }
+
+  LOG(INFO) << "Cache stats:\n" << memoryCacheStats.toString();
+}
+
+void PeriodicStatsReportDaemon::reportArbitratorStats() {
+  if (arbitrator_ == nullptr) {
+    return;
+  }
+  const auto updatedArbitratorStats = arbitrator_->stats();
+  VELOX_CHECK_GE(updatedArbitratorStats, prevArbitratorStats_);
+  const auto deltaArbitratorStats =
+      updatedArbitratorStats - prevArbitratorStats_;
+  REPORT_IF_NOT_ZERO(
```

Comment:
Can you check SharedArbitrator code? I think most of metrics have already been reported. And we only need to report the average arbitrator free spaces here? Thanks!

Reply:
synced offline. Will followup remove the duplicated reported metrics in arbitrator

Reply:
Let's do in this PR? We can't duplicate the report from velox side.